e2e_context_test.go

```diff
@@ -0,0 +1,305 @@
+package httpexpect
```

Comment:
New tests are great except one problem: they assume that one goroutine which sleeps for N seconds is always guaranteed to wake up before another goroutine that sleeps for N+1 seconds.

However, it's not really guaranteed. If tests are running on a loaded machine, on slow CI workers, inside a VM, there could be random delays in random goroutines, and test wont be 100% reliable, i.e. they can randomly fail from time to time.

It seems in this case it's relatively straightforward to make tests 100% reliable: instead of using time.Sleep() in request handler, we can use blocking channels.

Reply:
Addressed with  89d567c66fdc5afc27340ce88640380b3f968b2c